x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -946,6 +946,52 @@ protected static String typeErrorMessage(boolean includeOrdinal, List<Set<DataTy
             ),
             "boolean or datetime or numeric or string"
         ),
+        // to_int, to_long, to_double
+        Map.entry(
+            Set.of(
+                DataTypes.BOOLEAN,
+                EsqlDataTypes.COUNTER_INTEGER,
+                DataTypes.DATETIME,
+                DataTypes.DOUBLE,
+                DataTypes.INTEGER,
+                DataTypes.KEYWORD,
+                DataTypes.LONG,
+                DataTypes.TEXT,
+                DataTypes.UNSIGNED_LONG,
+                DataTypes.NULL
+            ),
+            "boolean or counter_integer or datetime or numeric or string"
+        ),
+        Map.entry(
+            Set.of(
+                DataTypes.BOOLEAN,
+                EsqlDataTypes.COUNTER_LONG,
+                DataTypes.DATETIME,
+                DataTypes.DOUBLE,
+                DataTypes.INTEGER,
+                DataTypes.KEYWORD,
+                DataTypes.LONG,
+                DataTypes.TEXT,
+                DataTypes.UNSIGNED_LONG,
+                DataTypes.NULL
+            ),
+            "boolean or counter_long or datetime or numeric or string"
+        ),
+        Map.entry(
+            Set.of(
+                DataTypes.BOOLEAN,
+                EsqlDataTypes.COUNTER_DOUBLE,
+                DataTypes.DATETIME,
+                DataTypes.DOUBLE,
+                DataTypes.INTEGER,
+                DataTypes.KEYWORD,
+                DataTypes.LONG,
+                DataTypes.TEXT,
+                DataTypes.UNSIGNED_LONG,
+                DataTypes.NULL
+            ),
+            "boolean or counter_double or datetime or numeric or string"
+        ),
```

Comment:
I've grown to hate this `Map`. I made it, but I dislike it....

Reply:
You've done the right thing to it, but I'm sorry you had to.

Reply:
There's another version of `errorsForCasesWithoutExamples` which takes another parameter that outputs a matcher for hte expected type. You could use that rather than changing the map.